### PR TITLE
BUGFIX: broken/missing Fortran code/unit tests

### PR DESCRIPTION
**Pull request by scivision**
The Fortran self-tests were broken even with all default settings with Gfortran, due to missing methods in compilers/fortran.py. Added those methods and cleanup/fix/modernize Fortran self-tests.

`compiler.run()` for Fortran has also been broken for a while, now fixed.

### CODE REVIEW

File: test cases/fortran/10 find library/meson.build [hunk 1]
```diff
@@ -3,7 +3,7 @@ project('find fortran library', 'fortran')
 fortranc = meson.get_compiler('fortran')
 
 sources = ['main.f90', 'gzip.f90']
-zlib = fortranc.find_library('z')
```
Comment (jpakkane) [inline test cases/fortran/10 find library/meson.build]: This used find_library on purpose to test that it is working (which is why the test directory is called `10 find library`).

File: test cases/fortran/5 static/meson.build [hunk 2]
```diff
@@ -1,6 +1,6 @@
-project('try-static-library', 'fortran')
+project('try-static-library', 'fortran', default_options : ['default_library=static'])
 
-static_hello = static_library('static_hello', 'static_hello.f90')
+static_hello = library('static_hello', 'static_hello.f90')
```
Comment (jpakkane) [inline test cases/fortran/5 static/meson.build]: What was the point of changing this (and the shared ilb test)? There is no noticeable advantage to doing this via `library` since it is explicitly testing that `static_library` works. It is clearer and more explicit to to just use `static_library`.

File: test cases/fortran/9 cpp/fortran.f90 [hunk 3]
```diff
@@ -0,0 +1,11 @@
+function fortran() bind(C)
```
Comment (jpakkane) [inline test cases/fortran/9 cpp/fortran.f90]: This test file had the suffix `.f` specifically to test that files with such extension get compiled. Are there compilers that do not handle `.f` files or why was this renamed?

File: test cases/fortran/9 cpp/main.cpp [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 extern "C" double fortran();
 
-int main(int, char**) {
-    std::cout << "FORTRAN gave us this number: " << fortran() << '\n';
-    return 0;
+int main(void) {
+    std::cout << "FORTRAN gave us this number: " << fortran() << std::endl;
```
Comment (jpakkane) [inline test cases/fortran/9 cpp/main.cpp]: These changes are not according to contemporary recommended C++ code. Leaving out argument names is the standard for not used arguments, and `std::endl` is recommended against unless it is explicitly needed (which is not the case here).

**Comment by jpakkane** [comment]
Please be careful with commits. For example this one has one commit that has a lot of travis changes and later a different commit that reverts all of them. They make reviewing the code hard and thus should not be there at all.

File: test cases/fortran/5 static/meson.build (hunk 2 above)
Comment (scivision) [reply test cases/fortran/5 static/meson.build]: reverted

File: test cases/fortran/10 find library/meson.build (hunk 1 above)
Comment (scivision) [reply test cases/fortran/10 find library/meson.build]: reverted

**Comment by scivision** [comment]
Yes the Fortran tests were quite broken, and I decided against adding Travis-based  Fortran CI for instead future Docker-based test changes.
